### PR TITLE
Render ECE button behind feature flag

### LINKED ISSUE #3402: [ECE] Integrate express buttons with ECE behind a feature flag
Integrate express buttons with [ECE](https://docs.stripe.com/elements/express-checkout-element) behind a new feature flag.
 
### Acceptance Criteria
- Create a new feature flag for ECE
- New buttons are displayed on the checkout page when ff is enabled.
- New buttons are hidden from the checkout page when ff is disabled.

### Excluded
- Successful payment processing is not a requirement of this ticket. It will be handled by https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3403 and https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3404

### PR DESCRIPTION
**Pull request by Mayisha**
Fixes #3402 

## Changes proposed in this Pull Request:
- Updated the `"@stripe/react-stripe-js"` version to the latest. Otherwise ECE buttons are not loaded by Stripe.
- Introduced a new option `_wcstripe_feature_ece` as the feature flag.
- In the block checkout page, render the ECE component when the feature flag is enabled. If feature flag is disabled then render the PRB component.
- This PR only displays the ECE button, no functionality has been added yet.
- ECE button is rendered on block checkout only. 

## Testing instructions
- Run `npm install` in this branch.
- Enable the ECE feature flag (return `true` from `is_stripe_ece_enabled` function or in your database add an option `_wcstripe_feature_ece` and set `yes` as the value)
- Go to the block checkout page and confirm that the ECE button is rendered.

Screenshot 2024-09-05 at 1 42 29 AM

- Disable the ECE feature flag.
- Confirm that the PRB button is rendered on both the block and shortcode checkout page. 

Screenshot 2024-09-05 at 1 51 48 AM

### CODE REVIEW

File: client/blocks/express-checkout/express-checkout.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* global wc_stripe_payment_request_params */
+
+import React from 'react';
+import { Elements, ExpressCheckoutElement } from '@stripe/react-stripe-js';
+
+export const ExpressCheckout = ( { stripe } ) => {
+	const options = {
```
Comment (wjrosa) [inline client/blocks/express-checkout/express-checkout.js:8]: Are these hardcoded options intended?

**Review by wjrosa — CHANGES_REQUESTED** [review]
Hum, for some reason it is not working properly for me.

When using localhost, I get a Sripe error saying that it is not possible to preview Express payment methods without https.

With a JN site, I get this error instead:
Screenshot 2024-09-05 at 13 21 22

Looking at the code, this is happening here:
```
if ( wc_stripe_payment_request_params.stripe.is_ece_enabled ) {
	// Register Express Checkout Element.
...
```
`wc_stripe_payment_request_params` is not defined here. Maybe something is missing to load this global variable 🤔

File: client/blocks/express-checkout/express-checkout.js (hunk 1 above)
Comment (Mayisha) [reply client/blocks/express-checkout/express-checkout.js:8]: Yes. This is just for this initial PR. We will replace it with real values in the later PRs.

**Comment by Mayisha** [comment]
Hi @wjrosa 

> wc_stripe_payment_request_params is not defined here. Maybe something is missing to load this global variable

I have fixed this in [122db02](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3420/commits/122db025ef7bf910bab4e27b855f40109d6e95dc).

**Review by wjrosa — APPROVED** [review]
The fix did the trick! Working as expected now:

> * Go to the block checkout page and confirm that the ECE button is rendered.

https://github.com/user-attachments/assets/4b8bea65-8d61-49fe-8f58-f9e85b307338

> * Confirm that the PRB button is rendered on both the block and shortcode checkout page.

https://github.com/user-attachments/assets/e6ac8b45-d120-4641-98e0-5fd8b0949bdd

https://github.com/user-attachments/assets/0f8cb4d9-eeb8-4309-9509-e9a4576a67d1

**Pull request merged**